From 580de0f38685264e4008878d4bc8847d0788ea3f Mon Sep 17 00:00:00 2001
From: Paul Sokolovsky <paul.sokolovsky@linaro.org>
Date: Thu, 7 Feb 2019 20:57:09 +0300
Subject: [PATCH] net: socket: Add zsock_freeaddrinfo()

The convention is that implementation of any Socket API function
lives in zsock_*(), and then it's optionally aliased to bare POSIX
name if requested. This convention must be followed to get proper
integration into the general POSIX subsystem.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>
---
 include/net/socket.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/net/socket.h b/include/net/socket.h
index 7583428254..b7998c3d5c 100644
--- a/include/net/socket.h
+++ b/include/net/socket.h
@@ -192,6 +192,11 @@ int ztls_setsockopt(int sock, int level, int optname,
 
 #endif /* defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS) */
 
+static inline void zsock_freeaddrinfo(struct zsock_addrinfo *ai)
+{
+	free(ai);
+}
+
 #if defined(CONFIG_NET_SOCKETS_POSIX_NAMES)
 #define pollfd zsock_pollfd
 #if !defined(CONFIG_NET_SOCKETS_OFFLOAD)
@@ -334,7 +339,7 @@ static inline int getaddrinfo(const char *host, const char *service,
 
 static inline void freeaddrinfo(struct zsock_addrinfo *ai)
 {
-	free(ai);
+	zsock_freeaddrinfo(ai);
 }
 
 #define addrinfo zsock_addrinfo
-- 
2.20.1

